core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/internal/Utils.java

```diff
@@ -103,7 +103,10 @@ public static String getAppNameFromRequest(Request request) {
             //if the request from jaxapps, getting appName from uri
             temp = uri.substring(uri.indexOf(JAX_APP_PATTERN) + 12);
         } else {
-            //if ST request getting the appName from the contex
+            //if ST request getting the appName from the context
+            if (request.getContext() == null) {
+                return null;
```

Comment:
Is it required to return null, shall we retrieve the appName from the context if the `request.getContext() != null`?

Reply:
Closing the PR as the check is not needed.